lib/css_parser/rule_set.rb

```diff
@@ -647,20 +652,31 @@ def parse_declarations!(block) # :nodoc:
       return unless block
 
       continuation = nil
-      block.split(/[;$]+/m).each do |decs|
-        decs = (continuation ? continuation + decs : decs)
-        if decs =~ /\([^)]*\Z/ # if it has an unmatched parenthesis
-          continuation = "#{decs};"
-        elsif (matches = decs.match(/\s*(.[^:]*)\s*:\s*(?m:(.+))(?:;?\s*\Z)/i))
-          # skip end_of_declaration
-          property = matches[1]
-          value = matches[2]
+      block.split(SEMICOLON) do |decs|
+        decs = (continuation ? "#{continuation};#{decs}" : decs)
+        if unmatched_parenthesis?(decs)
+          continuation = decs
+        else
+          colon = decs.index(COLON)
+          next if colon.nil?
+
+          property = decs[0, colon]
+          value = decs[colon + 1, decs.length - colon - 1]
+          property.strip!
+          value.strip!
+          next if property.empty? || value.empty?
+
           add_declaration!(property, value)
           continuation = nil
         end
       end
     end
 
+    def unmatched_parenthesis?(declarations)
```

Comment:
reads very nicely!

maybe more descriptive:
```suggestion
    def unmatched_open_parenthesis?(declarations)
```

should avoid `.nil?` for [performance reasons](https://stackoverflow.com/questions/1972266/obj-nil-vs-obj-nil)

```
(lparen_index = declarations.index(LPAREN)) && !declarations.index(RPAREN, lparen_index)
```